src/games/strategy/engine/framework/system/HttpProxy.java

```diff
@@ -0,0 +1,176 @@
+        Integer.parseInt(proxyPortArgument);
+        proxyPort = proxyPortArgument;
+      } catch (final NumberFormatException e) {
+        ClientLogger.logQuietly(e);
+      }
+    }
+    if (proxyHost != null || proxyPort != null) {
+      setProxy(proxyHost, proxyPort, ProxyChoice.USE_USER_PREFERENCES);
+    }
+    final Preferences pref = Preferences.userNodeForPackage(GameRunner.class);
+    final ProxyChoice choice = ProxyChoice.valueOf(pref.get(PROXY_CHOICE, ProxyChoice.NONE.toString()));
+    if (choice == ProxyChoice.USE_SYSTEM_SETTINGS) {
+      setToUseSystemProxies();
+    } else if (choice == ProxyChoice.USE_USER_PREFERENCES) {
+      final String host = pref.get(PROXY_HOST, "");
+      final String port = pref.get(PROXY_PORT, "");
+      if (host.trim().length() > 0) {
+        System.setProperty(HTTP_PROXYHOST, host);
+      }
+      if (port.trim().length() > 0) {
+        System.setProperty(HTTP_PROXYPORT, port);
+      }
+    }
+  }
+
+
+
+
+  public static void setProxy(final String proxyHost, final String proxyPort, final ProxyChoice proxyChoice) {
+    final Preferences pref = Preferences.userNodeForPackage(GameRunner.class);
+    final ProxyChoice choice;
+    if (proxyChoice != null) {
+      choice = proxyChoice;
+      pref.put(PROXY_CHOICE, proxyChoice.toString());
+    } else {
+      choice = ProxyChoice.valueOf(pref.get(PROXY_CHOICE, ProxyChoice.NONE.toString()));
+    }
+    if (proxyHost != null && proxyHost.trim().length() > 0) {
+      // user pref, not system properties
+      pref.put(PROXY_HOST, proxyHost);
+      if (choice == ProxyChoice.USE_USER_PREFERENCES) {
+        System.setProperty(HTTP_PROXYHOST, proxyHost);
+      }
+    }
+    if (proxyPort != null && proxyPort.trim().length() > 0) {
+      try {
+        Integer.parseInt(proxyPort);
+        // user pref, not system properties
+        pref.put(PROXY_PORT, proxyPort);
+        if (choice == ProxyChoice.USE_USER_PREFERENCES) {
+          System.setProperty(HTTP_PROXYPORT, proxyPort);
+        }
+      } catch (final NumberFormatException e) {
+        ClientLogger.logQuietly(e);
+      }
+    }
+    if (choice == ProxyChoice.NONE) {
+      System.clearProperty(HTTP_PROXYHOST);
+      System.clearProperty(HTTP_PROXYPORT);
+    } else if (choice == ProxyChoice.USE_SYSTEM_SETTINGS) {
+      setToUseSystemProxies();
+    }
+    if (proxyHost != null || proxyPort != null || proxyChoice != null) {
+      try {
+        pref.flush();
+        pref.sync();
+      } catch (final BackingStoreException e) {
+        ClientLogger.logQuietly(e);
+      }
+    }
+  }
+
+  private static void setToUseSystemProxies() {
+    final String javaNetUseSystemProxies = "java.net.useSystemProxies";
+    System.setProperty(javaNetUseSystemProxies, "true");
+    List<java.net.Proxy> proxyList = null;
+    try {
+      final ProxySelector def = ProxySelector.getDefault();
+      if (def != null) {
+        proxyList = def.select(new URI("http://sourceforge.net/"));
```

Comment:
Only moving code here. I'm very willing to consider PRs into this branch to further improve things